hoomd/md/AnisoPotentialPair.h

```diff
@@ -285,12 +293,12 @@ AnisoPotentialPair<aniso_evaluator>::AnisoPotentialPair(std::shared_ptr<SystemDe
                       m_rcutsq.getNumElements() * sizeof(Scalar),
                       cudaMemAdviseSetReadMostly,
                       0);
-        cudaMemAdvise(m_params.get(),
-                      m_params.getNumElements() * sizeof(param_type),
+        cudaMemAdvise(m_params.data(),
```

Comment:
I found this while debugging performance issues with @tcmoore3 on #1138. `m_rcutsq` is in unified memory when `allConcurrentManagedAccess()` returns `true`, but `m_params` and `m_shape_params` always in unified memory whenever running on the GPU.

Adjust this block accordingly:
```
if (isCUDAEnabled)
    {
    memadvise(m_params)
    memadvise(m_shape_params)
    prefetch(m_params)
    prefetch(m_shape_params)
    
    if (allConcurrentManagedAccess)
        {
        memadvise(m_rcutsq)
        prefetch(m_rcutsq)
        }
    }
````